pkg/server/in_cluster_application.go

```diff
@@ -0,0 +1,46 @@
+package server
+
+import (
+	"github.com/codeready-toolchain/registration-service/pkg/application"
+	"github.com/codeready-toolchain/registration-service/pkg/application/service"
+	servicecontext "github.com/codeready-toolchain/registration-service/pkg/application/service/context"
+	"github.com/codeready-toolchain/registration-service/pkg/application/service/factory"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/kubeclient"
+	"k8s.io/client-go/rest"
+)
+
+// NewInClusterApplication creates a new in-cluster application with the specified configuration and options.  This
+// application type is intended to run inside a Kubernetes cluster, where it makes use of the rest.InClusterConfig()
+// function to determine which Kubernetes configuration to use to create the REST client that interacts with the
+// Kubernetes service endpoints.
+func NewInClusterApplication(config configuration.Configuration, options ...factory.Option) (application.Application, error) {
+	app := new(InClusterApplication)
```

Comment:
do we really need to create this here? if we error out on line 22 or 27 this was created and never used

Reply:
Good point, I've moved it further down.